### PR TITLE
Fold Conv-Bn

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100653

Adds Conv-BN folding to inductor freezing. One thing that's a little awkward now is we'll want different decompositions to run depending on if we are in the inference compiler. For now, I require that you run with torch.no_grad() so we can detect if no gradients are required before calling aot_autograd.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov @chenyang78 @soumith @desertfire

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

### CODE REVIEW

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

**Review by jansel — APPROVED** [review]
